Java/src/main/java/com/gildedrose/item/CustomisedItemFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.gildedrose.item;
+
+public class CustomisedItemFactory {
+
+    public final static String SULFURAS = "Sulfuras, Hand of Ragnaros";
+    public final static String BRIE = "Aged Brie";
+    public final static String BACKSTAGE_PASSES_ITEM = "Backstage passes to a TAFKAL80ETC concert";
+    public final static String CONJURED_ITEM = "Conjured";
+
+    public CustomisedItem customiseItem(Item item) {
+        if (item.name.equals(SULFURAS)) {
```

Comment:
When you see a lot of if/else statements it can be a sign to use the Strategy pattern. There are trade off's of course. Perhaps research it and see whether you think the trade off's are worth it or not.

Reply:
Ok, will do that! The Strategy pattern is on my list for possible blog posts...

Reply:
Maybe too complicated in this kind of case but in general when having loads of if/else or case statements mapping from one thing to another can also consider actually using a Map to store the relationship e.g. BRIE -> AgedBrie::new. The function then just becomes a lookup with default in the Map.

Reply:
I made a refactoring in `replaces if-statements with Map in Factory Class` commit (see below), is that what you meant Phil? I replaced all the `if`-statements with a Map in the Factory Class